### PR TITLE
Add fuzziness to `knife ssh` queries

### PR DESCRIPTION
**Pull request by patcon**
PR for #3526.

### CODE REVIEW

File: lib/chef/knife/ssh.rb [hunk 1]
```diff
@@ -204,6 +207,12 @@ def search_nodes
         list
       end
 
+      def fuzzify_query
+        if @query !~ /:/
+          @query = "tags:*#{@query}* OR roles:*#{@query}* OR fqdn:*#{@query}* OR addresses:*#{@query}*"
```
Comment (patcon) [inline lib/chef/knife/ssh.rb]: Should this be "weaker"? Possibilities:

1. Remove asterisks for both sides.
2. fqdn only? fqdn + name?
3. Keep as is, but have a confirmation prompt for what fqdn we're running  on. (Can be explicitly skipped with `--yes` flag.)

cc: @lamont-granquist

**Comment by sawanoboly** [comment]
:+1:

File: lib/chef/knife/ssh.rb (hunk 1 above)
Comment (lamont-granquist) [inline lib/chef/knife/ssh.rb]: It needs to be the same as knife search node.

**Comment by patcon** [comment]
Also, my ruby force is apparently weak here -- any suggestions on how I could pull `#fuzzify_query` from `lib/chef/knife/search`? (this is where you all probably judge me :)

**Comment by patcon** [comment]
Also, went searching for tests that cover this part, but doesn't seem we test this level, so given that I'm not comfortable enough with starting my own test approach from scratch, I think this is just waiting on feedback on the above, then it's good to merge as far as I'm aware.

**Comment by lamont-granquist** [comment]
So, what I'd really like to see is knife ssh and knife node search share exactly the same code and call the same method that does fuzziness.  Eliminate code duplication and eliminate the chances that we patch in one place and not the other.

**Comment by lamont-granquist** [comment]
We can't merge this without considering the problem holistically, so :-1: for now.

**Comment by coderanger** [comment]
Agree, and for the future the Manage code has a similar fuzzifier and should make sure that uses the same algo.

**Comment by patcon** [comment]
Totally fair. Closing for now

**Pull request closed without merging**